build.rs

```diff
@@ -535,9 +525,12 @@ fn compile(p: &Path, target: &Target, warnings_are_errors: bool, out_dir: &Path)
     }
 }
 
-fn obj_path(out_dir: &Path, src: &Path, obj_ext: &str) -> PathBuf {
+fn obj_path(out_dir: &Path, src: &Path) -> PathBuf {
     let mut out_path = out_dir.join(src.file_name().unwrap());
-    assert!(out_path.set_extension(obj_ext));
+    // To eliminate unnecessary conditional logic, use ".o" as the extension,
+    // even when the compiler (e.g. MSVC) would normally use something else
+    // (e.g. ".obj"). cc-rs seems to do the same.
+    assert!(out_path.set_extension("o"));
```

Comment:
I hope this is fine to do. When I was trying to compile *ring* for Windows AArch64, clang refused to accept any extension other than `.S` for assembly files 🤷‍♂️ Hopefully MSVC is not so opinionated.

Reply:
> I hope this is fine to do. When I was trying to compile _ring_ for Windows AArch64, clang refused to accept any extension other than `.S` for assembly files 🤷‍♂️ Hopefully MSVC is not so opinionated.

In GCC/clang land, ".S" means "use the preprocessor, ".s" means "don't use the preprocessor, so that's kind of a different issue.